### ISSUE TITLE
Encoding works on pointer but not on struct directly

### ISSUE THREAD

**Issue by jpninanjohn** [go-yaml/yaml#518]
We were looking to insert comments into a yaml through golang code. Currently it works, but only if the field is given as a pointer(second example). If not given as a pointer(first example) it does not fail but renders the entire yaml.Node{} struct as a yaml.

## notworking.go
```

package main

import (
    "bytes"
    "fmt"
    "gopkg.in/yaml.v3"
)

func main() {

    type Person struct {
        Name    string
        Address yaml.Node
    }

    address := yaml.Node{
        Kind:        yaml.ScalarNode,
        LineComment: "A line comment",
        HeadComment: "A Head comment",
        Value:       "Address 1",
        Style:       yaml.DoubleQuotedStyle,
    }

    person := Person{
        Name:    "Leo",
        Address: address,
    }

    var buffer bytes.Buffer
    err := yaml.NewEncoder(&buffer).Encode(person)
    if err != nil {
        fmt.Println(err)
    }

    fmt.Print(buffer.String())

}
```
Output:
$ go run main.go
name: Leo
address:
    kind: 8
    style: 2
    tag: ""
    value: Address 1
    anchor: ""
    alias: null
    content: []
    headcomment: A Head comment
    linecomment: A line comment
    footcomment: ""
    line: 0
    column: 0
```

## working.go:

`package main

import (
    "bytes"
    "fmt"
    "gopkg.in/yaml.v3"
)

func main() {

    type Person struct {
        Name    string
        Address *yaml.Node
    }

    address := &yaml.Node{
        Kind:        yaml.ScalarNode,
        LineComment: "A line comment",
        HeadComment: "A Head comment",
        Value:       "Address 1",
        Style:       yaml.DoubleQuotedStyle,
    }

    person := Person{
        Name:    "Leo",
        Address: address,
    }

    var buffer bytes.Buffer
    err := yaml.NewEncoder(&buffer).Encode(person)
    if err != nil {
        fmt.Println(err)
    }

    fmt.Print(buffer.String())

}`

Output: ```
 go run main.go
name: Leo
address: "Address 1" # A line comment
# A Head comment
```

Is this expected? Or it a bug?

**Comment by siredwin** [comment]
How did you fix it? I have the same issue.

**Comment by jpninanjohn** [comment]
I used the working code. There was no limitation for me to use a non pointer. Just that it took me some time to get to the problem.

**Comment by niemeyer** [comment]
This has been fixed in early May with a81ed60d. Please update your code and it should work. If it doesn't, please let me know.
